### ISSUE TITLE
Add support for multiple different blogs

### ISSUE THREAD

**Issue by kemalkastrataj** [lloc/Multisite-Language-Switcher#400]
**Describe the bug**
The Multisite Language Switcher (MSLS) plugin currently lacks the functionality to associate different sets of sites within a multisite or multi-network setup. In scenarios where multiple groups of sites exist—such as a client with a coaching site spanning two associated sites (where MSLS works correctly) and a third, entirely unrelated site—the plugin cannot accommodate separate translation associations.

For example, if I were to add a German version for the unrelated third site, MSLS would incorrectly attempt to treat all four sites (the two coaching sites and the unrelated site plus its translation) as being in the same translation group. Currently, there's no way to group or separate translations for different site sets within a network, which limits MSLS’s flexibility for complex site structures.

**Comment by lloc** [comment]
Thanks for your input on this particular scenario. Normally, you can decide the sites that you want to tie together by assigning them to a specific user. There is the field “Reference User” which is helpful for this task. For even more accentuated separation of sites in a multisite, you could also use a plugin called “WP Multi Network”. This plugin got not much love during the last major updates of WordPress, but I have plans to work on it too.

**Comment by martin-neumann-gurus** [comment]
I just tested to use the reference user in order to divide two different blogs with in a network. One network I have set with the super admin and the other one, with a newly created admin that had no super admin privileges. The network with the new admin worked fine, but the one that used the super admin, was including the other network as well. Do I need to use two users that have no super admin privileges in order to make that work?

**Comment by kemalkastrataj** [comment]
I haven't tested this yet, but just based on theory, a dropdown within the site's MSLS settings section would be better. Maybe something like a parent site, so you can designate site X as the parent of all these others.

Or do you think it would be better to group them by user?

**Comment by lloc** [comment]
You can use any user as “Reference User” to tie the websites together. This can be even a subscriber because the plugin uses this info only for that specific info.

The “Reference User” setting is a dropdown and I believe it is ok like it is since you can decide on a website level who is in or out.

**Comment by martin-neumann-gurus** [comment]
Yes I can see what the reference user is supposed to do. But in my tests it did not quite worked out as it should. Depending on the user you choose (Maybe depending on the privilleges of the user) it includes more blogs than it should. It is not quite working right yet. If you want we can connect and I will show you what is going on here at my installation.

**Comment by Slavomir9** [comment]
<p data-path-to-node="3">Dear Developer,</p>
<p data-path-to-node="4">I am currently running a WordPress Multisite network with 4 sites in two languages, focused on two topics: "Cars" and "IT" (set up for testing purposes):</p>
<ul data-path-to-node="5">
<li>
<p data-path-to-node="5,0,0"><strong data-path-to-node="5,0,0" data-index-in-node="0">sk-auta</strong> (Slovak - Cars)</p>
</li>
<li>
<p data-path-to-node="5,1,0"><strong data-path-to-node="5,1,0" data-index-in-node="0">en-cars</strong> (English - Cars)</p>
</li>
<li>
<p data-path-to-node="5,2,0"><strong data-path-to-node="5,2,0" data-index-in-node="0">sk-it</strong> (Slovak - IT)</p>
</li>
<li>
<p data-path-to-node="5,3,0"><strong data-path-to-node="5,3,0" data-index-in-node="0">en-it</strong> (English - IT)</p>
</li>
</ul>
<p data-path-to-node="6">Using the <strong data-path-to-node="6" data-index-in-node="10">Multisite Language Switcher</strong> plugin, I am able to link articles between different language versions. However, the current interface is not very user-friendly for this specific setup.</p>
<p data-path-to-node="7">The main issue is that the interface is quite cluttered. For example, when I am in the post list or editing an article on the <strong data-path-to-node="7" data-index-in-node="126">en-cars</strong> site, the plugin gives me the option to pair the post with sites from completely different topics (e.g., it shows pairing options for both <strong data-path-to-node="7" data-index-in-node="273">sk-auta</strong> and <strong data-path-to-node="7" data-index-in-node="285">sk-it</strong>).</p>
<p data-path-to-node="8">I have tested this on a live WordPress Multisite installation as well, but the problem persists. I haven't been able to find anything in the documentation regarding how to restrict pairing to specific "site pairs" within the network.</p>
<p data-path-to-node="9"><strong data-path-to-node="9" data-index-in-node="0">My Goal:</strong> If I am on the <strong data-path-to-node="9" data-index-in-node="24">en-cars</strong> site, I would like to see only the option to link or create a version for <strong data-path-to-node="9" data-index-in-node="106">sk-auta</strong>. I want to hide all other unrelated sites (like <strong data-path-to-node="9" data-index-in-node="162">sk-it</strong> or <strong data-path-to-node="9" data-index-in-node="171">en-it</strong>) from the selection to keep the workflow clean and prevent errors.</p>
<p data-path-to-node="10"><strong data-path-to-node="10" data-index-in-node="0">My Questions:</strong></p>
<ol start="1" data-path-to-node="11">
<li>
<p data-path-to-node="11,0,0">Are there any existing settings to define "site relationships" or pairs within the network to filter these options?</p>
</li>
<li>
<p data-path-to-node="11,1,0">If this functionality is not currently included in Multisite Language Switcher, would it be possible to add a feature that allows pairing specific websites together?</p>
</li>
</ol>
<p data-path-to-node="12">I would be very grateful for such an addition, as it would significantly improve the user experience for multi-topic networks.</p>
<p data-path-to-node="13">Thank you very much in advance for your time and help.</p>
<p data-path-to-node="14">Best regards,</p>

<img width="1890" height="730" alt="Image" src="https://github.com/user-attachments/assets/8ccecbfa-3eb9-4214-a330-77e904dc8775" />
<img width="1934" height="733" alt="Image" src="https://github.com/user-attachments/assets/9e925d15-821c-4245-af65-2e4a8e9e15e6" />
<img width="1910" height="874" alt="Image" src="https://github.com/user-attachments/assets/f07e0ca0-6163-4fdc-9a88-e53025ed8127" />
<img width="1273" height="700" alt="Image" src="https://github.com/user-attachments/assets/f9e775af-1fdb-4781-8d47-4fb5cd0ccdfa" />
<img width="1032" height="581" alt="Image" src="https://github.com/user-attachments/assets/e9f8c67d-507f-4fa1-8c9d-2864ca7e4f24" />

**Comment by lloc** [comment]
Hello, the usual workaround is that you define 2 reference users (they can be even subscribers).

I cite the docs here ;)

> **Reference user**
> 
> This is an important but somewhat overseen field.
> 
> The easiest (but not the only one) way to connect websites is by the user. There is no problem if you have just one user that accesses all the websites. But this option is useful if you want to connect a subset or if you have more than 1 admin.
> 
> Just set who is the reference for the connections and you are done.

Source: https://msls.co/user-docs/settings-in-the-administration-menu/
